tests/reframe/testlib/applications/gromacs/__init__.py

```diff
@@ -0,0 +1,67 @@
+import os
+import reframe as rfm
+import reframe.utility.sanity as sn
+
+# 3.5.0 Required for using current_partition.processor
+#@rfm.required_version('>=3.5.0')
+class Gromacs(rfm.RunOnlyRegressionTest, pin_prefix=True):
+    '''Gromacs benchmark based on Prace Benchmark Suite GROMACS case A.
+
+    Derived tests must specify the variables ``num_tasks``, ``num_tasks_per_node``, ``num_cpus_per_task``, ``nsteps`` and ``modules``.
+    Note that a sufficiently large ``nsteps`` needs to be defined in order for GROMACS to pass the load balancing phase.
+    As a rough estimate: 10000 steps would generally be ok for 24 tasks, a 100000 steps for 240 tasks, etc.
+    '''
+
+    num_tasks = required
+    num_tasks_per_node = required
+    num_cpus_per_task = required
+    nsteps = required
+    modules = required
```

Comment:
Hm, fair point. My idea when designing this class was to create a library test for just running a GROMACS module. But indeed, nothing specific is in this library that would make this a 'run-only' test. The person deriving from the test might technically build GROMACS as part of the test, and this would still be a valid library test to derive from. I wouldn't recommend it, but you're right: there's no use in _forcing_ the use of a module here.

Reply:
No, wait, let me get back to this: I derived the library test from RunOnlyRegressionTest. Doesn't _that_ essentially make a module required?

Reply:
Not necessarily. One could have installed gromacs in the system manually. As long as the executable `gmx_mpi` is there, this test can run.

Reply:
Fair point. So used to managing things with modules that I sometimes forget what happens on non-HPC systems =)